### PR TITLE
Add ApplicationCRD Enabled ENV in Publisher deployment

### PR DESCRIPTION
**Pull request by mfaizanse**
**Description**

Changes proposed in this pull request:

- Add ApplicationCRD Enabled ENV in Publisher deployment

**Related issue(s)**

- https://github.com/kyma-project/eventing-manager/issues/94

### CODE REVIEW

File: pkg/k8s/client_test.go [hunk 1]
```diff
@@ -535,3 +538,99 @@ func Test_GetValidatingWebHookConfiguration(t *testing.T) {
...
+	}{
+		{
+			name:              "should return not found error when CRD is missing in k8s",
+			givenCRDName:      ApplicationCrdName,
+			wantNotFoundError: false,
+		},
+		{
+			name:              "should return correct CRD from k8s",
+			givenCRDName:      "non-existing",
+			wantNotFoundError: true,
```
Comment (marcobebway) [inline pkg/k8s/client_test.go:559]: I think the names of the test cases are switched.

File: pkg/k8s/types.go [hunk 2]
```diff
@@ -0,0 +1,10 @@
+package k8s
+
+const (
+	// ApplicationCrdName defines the CRD name for Application of application-connector module.
+	ApplicationCrdName string = "applications.applicationconnector.kyma-project.io"
+	// ApplicationKind defines the Kind name for Application of application-connector module.
+	ApplicationKind string = "Application"
+	// ApplicationAPIVersion defines the API version for Application of application-connector module.
+	ApplicationAPIVersion string = "applicationconnector.kyma-project.io/v1alpha1"
```
Comment (marcobebway) [inline pkg/k8s/types.go:9]: Please check if the Kyma application connector package have similar constants or not, so we can reuse them.

Comment (marcobebway) [reply pkg/k8s/types.go:9]: Sure

File: pkg/k8s/client_test.go (hunk 1 above)
Comment (mfaizanse) [reply pkg/k8s/client_test.go]: updated

**Review by marcobebway — APPROVED** [review]

**Pull request merged**